core/src/test/deposit_and_withdraw_e2e.rs

```diff
@@ -1552,3 +1555,244 @@ async fn mock_citrea_run_malicious_after_exit() {
 
     assert!(tx.output[0].value != config.protocol_paramset().operator_challenge_amount);
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn concurrent_deposits_and_withdrawals() {
```

Comment:
Small comment explaining what the test does would be appreciated since it contains ~300 lines of code.